DOCSP-30520: distinct usage example (#72)
mongodb/docs-rust: source/usage-examples/distinct.txt

.. _rust-distinct-usage:

==========================
List Distinct Field Values
==========================

> You can list the distinct values of a document field across a collection
> by calling the `distinct() <{+api+}/struct.Collection.html#method.distinct>`__
> method on a ``Collection`` instance.

Reviewer: This sentence was confusing to me. I think the "across a collection" part makes it less clear. It might be more clear worded something like this:

```suggestion
You can list the values of a field that are distinct from one another across a collection
by calling the `distinct() <{+api+}/struct.Collection.html#method.distinct>`__
method on a ``Collection`` instance.
```

Author: should I just remove the phrase "across a collection" instead? Is this implicit in the fact that you call the method on a `Collection` instance?

Reviewer: Yeah you're right, I think since it's pointed out that it's being called on `Collection` makes it implicit that it's "across the collection." Maybe removing "across the collection" and then phrasing it as "distinct field values" (like it is in the examples section) makes it feel more clear IMO. So:

"You can list the distinct field values of a document by calling the . . . "

Pass a field name as a parameter to the ``distinct()`` method to return

> the distinct values for that field. You can also pass a query filter as
> a parameter to find distinct values only in a subset of matched

Reviewer: Suggestion: The first time I read this I read it as "to find distinct values only..." where only is referring to "distinct values" rather than "only in a subset." (hope that explanation makes sense)  
 
```suggestion
the distinct values for that field. You can also pass a query filter as
a parameter to find distinct values from only a subset of matched
```

documents. To learn more about creating query filters, see the
:ref:`rust-query-guide` guide.

The ``distinct()`` method returns the list of distinct values as a ``Vec<Bson>``
type, a vector of `Bson <{+bson-api+}/enum.Bson.html>`__ values.

Example
-------

> This example finds distinct values for a field across the
> ``restaurants`` collection of the ``sample_restaurants`` database. The

Reviewer: Similar to my comment earlier, I think "for a field across the collection" makes the sentence a little harder to understand. 

```suggestion
This example finds values for a field that are distinct across the
``restaurants`` collection of the ``sample_restaurants`` database. The
```

Author: What about "distinct field values"

example uses a ``Restaurant`` struct that has ``cuisine`` and
``borough`` fields to model documents in the collection.

The following code finds distinct values of the ``borough`` field in
the subset of documents in which the value of the ``cuisine`` field

> is ``"Turkish"``.

Reviewer: ```suggestion
is ``"Turkish"``:
```

Author: will change in all merged UEs
